Java/Mazes/src/DrawMaze.java

```diff
@@ -0,0 +1,427 @@
+                } else if (p.equals(checkLeft)) {
+                    availableDirection.add(3);
+                } else if (p.equals(checkRight)) {
+                    availableDirection.add(4);
+                }
+            }
+
+            // If there is no adjacent loose point, remove the point from the source list
+            if (availableDirection.size() == 0) {
+                removePoint = true;
+            } else if (availableDirection.size() == 1) { // If there is only 1 direction to travel,
+                // add a wall in that direction and remove the source point from the source array
+                removePoint = true;
+                direction = availableDirection.get(0);
+            } else {
+                // If there is more than 1 adjacent point, randomly pick a direction
+                direction = availableDirection.get((int) (Math.random() * availableDirection.size()));
+            }
+
+            // Based on the chosen direction, draw a wall
+            if (direction == 1) { // DRAW UP
+                this.addVerticalWall((int) sourcePoint.getX(), (int) sourcePoint.getY() - 1, 1);
+                loose.remove(checkUp);
+                source.add(checkUp);
+            } else if (direction == 2) {  // DRAW DOWN
+                this.addVerticalWall((int) sourcePoint.getX(), (int) sourcePoint.getY(), 1);
+                loose.remove(checkDown);
+                source.add(checkDown);
+            } else if (direction == 3) {  // DRAW LEFT
+                this.addHorizontalWall((int) sourcePoint.getX() - 1, (int) sourcePoint.getY(), 1);
+                loose.remove(checkLeft);
+                source.add(checkLeft);
+            } else if (direction == 4) { //DRAW RIGHT
+                this.addHorizontalWall((int) sourcePoint.getX(), (int) sourcePoint.getY(), 1);
+                loose.remove(checkRight);
+                source.add(checkRight);
+            }
+
+            // If the source point has 0 or 1 adjacent points, remove it from the array
+            if (removePoint) {
+                source.remove(sourcePoint);
+            }
+        }
+        display();
+    }
+
+    /**
+     * Algorithm to generate a maze via Depth-First Search (DFS) until all cells have been visited This maze is created
+     * by generating a wall filled maze and then removing walls in a given path
+     */
+    private void maze2() throws NullPointerException {
+
+        ArrayList<Cell> cellGrid = new ArrayList<>();
+
+        // Generate a wall filled maze
+        for (int i = 0; i < mazeWidth; i++) {
+            for (int j = 0; j < mazeHeight; j++) {
+
+                addVerticalWall(i, j);
+                addHorizontalWall(i, j);
+
+                cellGrid.add(new Cell(i, j));
+            }
+        }
+
+        // Remove the wall on the opening
+        walls.remove(new Wall(0, 0, true));
+
+        // Create the stack to use for recursive pathfinding
+        Stack<Cell> stack = new Stack<>();
```

Comment:
IF you can think of a better name for this stack than "Stack" you won't need to add this comment.